src/inference_engines/mlc_engine.py

```diff
@@ -172,5 +172,10 @@ def __call__(
                 break
             self.cm._decode(generation_config=generation_config)
             out = self.cm._get_message()
+            # stops us from yielding half an emoji, which breaks
+            out = out.replace("�", "") 
```

Comment:
if that's the right character, this might be clearer

```suggestion
            out = out.replace("\N{Replacement Character}", "") 
```